pkg/kubelet/cm/dra/plugin/client.go

```diff
@@ -32,122 +32,157 @@ import (
 
 const PluginClientTimeout = 45 * time.Second
 
-// draPluginClient encapsulates all dra plugin methods.
-type draPluginClient struct {
-	pluginName string
-	plugin     *Plugin
-}
-
 func NewDRAPluginClient(pluginName string) (drapb.NodeClient, error) {
 	if pluginName == "" {
 		return nil, fmt.Errorf("plugin name is empty")
 	}
 
-	existingPlugin := draPlugins.Get(pluginName)
+	existingPlugin := draPlugins.get(pluginName)
 	if existingPlugin == nil {
 		return nil, fmt.Errorf("plugin name %s not found in the list of registered DRA plugins", pluginName)
 	}
 
-	return &draPluginClient{
-		pluginName: pluginName,
-		plugin:     existingPlugin,
-	}, nil
+	return existingPlugin, nil
 }
 
-func (r *draPluginClient) NodePrepareResources(
+func (p *plugin) NodePrepareResources(
 	ctx context.Context,
 	req *drapb.NodePrepareResourcesRequest,
 	opts ...grpc.CallOption,
-) (resp *drapb.NodePrepareResourcesResponse, err error) {
+) (*drapb.NodePrepareResourcesResponse, error) {
 	logger := klog.FromContext(ctx)
 	logger.V(4).Info(log("calling NodePrepareResources rpc"), "request", req)
-	defer logger.V(4).Info(log("done calling NodePrepareResources rpc"), "response", resp, "err", err)
 
-	conn, err := r.plugin.getOrCreateGRPCConn()
+	conn, err := p.getOrCreateGRPCConn()
 	if err != nil {
 		return nil, err
 	}
-	nodeClient := drapb.NewNodeClient(conn)
-	nodeClientOld := drapbv1alpha2.NewNodeClient(conn)
 
 	ctx, cancel := context.WithTimeout(ctx, PluginClientTimeout)
 	defer cancel()
 
-	resp, err = nodeClient.NodePrepareResources(ctx, req)
+fallback:
+	p.RLock()
+	fallbackVersion := p.fallbackVersion
+	p.RUnlock()
+	if fallbackVersion == "v1alpha2" {
+		response := v1alpha2NodePrepareResouce(ctx, conn, req.Claims)
```

Comment:
This call looks a bit inconsistent to me. When we fallback to `v1alpha2` we don't use nodeClient, but otherwise we do. Can we hide this complexity inside NodeClient implementation?

Reply:
I tried to make this part clearer and moved out the node client from the function. It is now created here before calling `v1alpha2NodePrepareResouce`. 

`Can we hide this complexity inside NodeClient implementation?` I am not sure to understand that part, you mean in the code generated from the proto file ?

Reply:
You're right, it can't be handled in the generated file. I'd propose to wrap details into functions for both supported versions. Currently this is a wrapper function `v1alpha2NodePrepareResouce` only for one of two cases.

Reply:
done in 17b634683020ed2ab31936ac3248cc8c9cda97b5